scripts/upgrade

```diff
@@ -1,25 +1,188 @@
+# NGINX CONFIGURATION
+#=================================================
+ynh_script_progression --message="Upgrading nginx web server configuration..." --weight=1
+
+# Create a dedicated nginx config
+ynh_add_nginx_config
+
+#=================================================
+# CREATE DEDICATED USER
+#=================================================
+ynh_script_progression --message="Making sure dedicated system user exists..." --weight=1
+
+# Create a dedicated user (if not existing)
+ynh_system_user_create --username="list"
+
+#=================================================
+# SPECIFIC UPGRADE
+#=================================================
+# TWEAKING ....
+#=================================================
+ynh_script_progression --message="Add YunoHost specific theme..." --weight=1
+
+# Custom template for listinfo
+cp ../sources/html/listinfo.html /var/lib/mailman/templates/en/
+
+#=================================================
+# POSTFIX CONFIGURATION
+#=================================================
+
+ynh_script_progression --message="Configuring postfix mail server..."
+
+# Add postfix configuration hook and regen postfix conf
+cp -R ../sources/hooks/conf_regen/98-postfix_mailman /usr/share/yunohost/hooks/conf_regen/
+yunohost tools regen-conf postfix
+
+#=================================================
+
+#=================================================
+# MODIFY A CONFIG FILE
+#=================================================
+ynh_script_progression --message="Upgrading mailman configuration file..." --weight=1
+
+### Verify the checksum of a file, stored by `ynh_store_file_checksum` in the install script.
+### And create a backup of this file if the checksum is different. So the file will be backed up if the admin had modified it.
+ynh_backup_if_checksum_is_different --file="/etc/mailman/mm_cfg.py"
+
+ynh_replace_string "__MAILMAN_DOMAIN__"   "$domain"              ../conf/mm_cfg.py
+ynh_replace_string "__MAILMAN_WEB_PATH__" "$path_url" ../conf/mm_cfg.py
+ynh_replace_string "__MAILMAN_LANGUAGE__" "$language"            ../conf/mm_cfg.py
+cp ../conf/mm_cfg.py /etc/mailman/mm_cfg.py
+
+# Recalculate and store the checksum of the file for the next upgrade.
+ynh_store_file_checksum --file="/etc/mailman/mm_cfg.py"
+
+#=================================================
+# GENERIC FINALIZATION
+#=================================================
+# SECURE FILES AND DIRECTORIES
+#=================================================
+
+# Set permissions on app files
+chown -hR list:list /var/lib/mailman/
+/var/lib/mailman/bin/check_perms -f > /dev/null
+chown list:www-data -R /var/lib/mailman/archives/
+
+#=================================================
+# INTEGRATE SERVICE IN YUNOHOST
+#=================================================
+ynh_script_progression --message="Integrating service in YunoHost..." --weight=1
+
+yunohost service add $app --description "A short description of the app" --log "/var/log/$app/$app.log"
```

Comment:
Description should be improved, + pretty sure that /var/log/mailman/mailman.log doesnt exist :s